flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/reconciler/sessionjob/SessionJobReconcilerTest.java

```diff
@@ -232,6 +232,218 @@ public void testStatelessUpgrade() throws Exception {
                 flinkService.listJobs());
     }
 
+    @Test
```

Comment:
I think this can be compacted such a way but that said in my previous comment these are passing all the time no matter if your fix is there or not:
```
    @ParameterizedTest
    @EnumSource(org.apache.flink.api.common.JobStatus.class)
    public void testCancelStatelessSessionJob(org.apache.flink.api.common.JobStatus fromJobStatus) throws Exception {
        FlinkSessionJob sessionJob = TestUtils.buildSessionJob();

        var readyContext = TestUtils.createContextWithReadyFlinkDeployment();

        reconciler.reconcile(sessionJob, readyContext);
        assertEquals(1, flinkService.listJobs().size());
        verifyAndSetRunningJobsToStatus(
                sessionJob,
                JobState.RUNNING,
                org.apache.flink.api.common.JobStatus.RECONCILING.name(),
                null,
                flinkService.listJobs());

        var statelessSessionJob = ReconciliationUtils.clone(sessionJob);
        var jobConfig = flinkService.listJobs().get(0).f2;

        // JobID must be equal.
        assertEquals(
                statelessSessionJob.getStatus().getJobStatus().getJobId(),
                flinkService.listJobs().get(0).f1.getJobId().toHexString());

        statelessSessionJob.getSpec().getJob().setUpgradeMode(UpgradeMode.STATELESS);
        statelessSessionJob.getSpec().getJob().setParallelism(2);
        reconciler.reconcile(statelessSessionJob, readyContext);

        statelessSessionJob
                .getStatus()
                .getJobStatus()
                .setState(fromJobStatus.name());
        flinkService.cancelSessionJob(statelessSessionJob, UpgradeMode.STATELESS, jobConfig);
        assertEquals(
                org.apache.flink.api.common.JobStatus.FINISHED,
                flinkService.listJobs().get(0).f1.getJobState());
        verifyJobState(statelessSessionJob, JobState.SUSPENDED, "FINISHED");
    }
```